api/kubeconfig: Generate the clusterRoleBinding for harvester-csi-driver

Vicente-Cheng opened the pull request:
related issues: #3052, #3467

We need to add related work when we create the `HARVESTER` cloud provider.
That will make the harvester-CSI-driver can check with the `StorageClass` on the host cluster.

Also need the UI-related work to change the API payload like the following:
(refer to this https://github.com/rancher/dashboard/blob/master/shell/edit/provisioning.cattle.io.cluster/rke2.vue#L1215)
```yaml
    csiClusterRoleName: `harvesterhci.io:csi-driver`
    clusterRoleName:    'harvesterhci.io:cloudprovider',
    namespace: <namespace>,
    serviceAccountName: <name>
```

cc @guangbochen, @bk201 

**Test Plan**
1. create 3 node cluster using the ISO included this PR
2. create guest cluster
3. check the `ClusterRole` and `ClusterRoleBinding` with following contents:

**ClusterRole**
```
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  annotations:
    meta.helm.sh/release-name: harvester
    meta.helm.sh/release-namespace: harvester-system
    objectset.rio.cattle.io/id: default-mcc-harvester-cattle-fleet-local-system
  creationTimestamp: "2023-12-18T15:55:44Z"
  labels:
    app.kubernetes.io/component: apiserver
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: harvester
    app.kubernetes.io/part-of: harvester
    app.kubernetes.io/version: master-85038796-dirty
    helm.sh/chart: harvester-0.0.0-master-85038796-dirty
    helm.sh/release: harvester
    objectset.rio.cattle.io/hash: e852fa897f5eae59a44b4bfe186aad80b10b94b3
  name: harvesterhci.io:csi-driver
  resourceVersion: "3961"
  uid: 9e3a5b4f-97b9-43a6-a304-4b75b73a2b35
rules:
- apiGroups:
  - storage.k8s.io
  resources:
  - storageclasses
  verbs:
  - get
  - list
  - watch
```

**ClusterRoleBinding**
```
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  creationTimestamp: "2023-12-19T03:23:39Z"
  name: default-rke2-01
  ownerReferences:
  - apiVersion: v1
    kind: ServiceAccount
    name: rke2-01
    uid: 25f3858d-b865-46b4-b7b9-072ac35b554b
  resourceVersion: "633316"
  uid: 11ebd4d9-6d78-460e-8da5-f036f9d8052c
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: harvesterhci.io:csi-driver
subjects:
- kind: ServiceAccount
  name: rke2-01
  namespace: default
```
Also, make sure the `ServiceAccount` information should be correct.

Vicente-Cheng committed:
api/kubeconfig: Generate the clusterRoleBinding for harvester-csi-driver

    We need to check the StorageClass on the host cluster, so we need
    to allow the guest cluster access host StorageClass resource.

    Add a new ClusterRole `harvesterhci.io:csi-driver` to allow
    read-related operation on the harvester-CSI-driver.

Signed-off-by: Vicente Cheng <vicente.cheng@suse.com>

guangbochen reviewed (commented):
~~Would prefer to add the storage permissions to the existing ClusterRole of `harvesterhci.io:cloudprovider` and create `clusterolebinding` instead of `rolebinding`, so that UI doesn't need to make any changes.~~
Additionally, we may need proper upgrade plans to migrate existing `rolebindings` to `clusterrolebindings` for existing guest k8s cluster.

Vicente-Cheng committed:
api/kubeconfig: fixed the clusterRole name

    - fixed the clusterRole name with creation because the rbac already
      configured it.

Signed-off-by: Vicente Cheng <vicente.cheng@suse.com>

ibrokethecloud reviewed (approved):
lgtm. thanks.

bk201 reviewed (approved):
LGTM! thanks!

starbops reviewed (approved):
LGTM, thank you!
